main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -684,6 +684,48 @@ Resources:
               Resource:
                 - !GetAtt [StudiesDb, Arn]
 
+    # IAM Role for the envStatusPollHandler Function
+  RoleEnvStatusPollHandler:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      Policies:
+        - PolicyName: db-environment-access
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action:
+                - dynamodb:GetItem
+                - dynamodb:Query
+                - dynamodb:Scan
+                - dynamodb:UpdateItem
+              Resource:
+                - !GetAtt [EnvironmentsScDb, Arn]
+        - PolicyName: db-scan
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action:
+                - dynamodb:Scan
+              Resource:
+                - !GetAtt [IndexesDb, Arn]
+                - !GetAtt [ProjectsDb, Arn]
+                - !GetAtt [AwsAccountsDb, Arn]
+        - PolicyName: assume-role
+          PolicyDocument:
+            Statement:
+              Action: 'sts:AssumeRole'
+              Effect: Allow
+              Resource: '*' # TODO: LOCK THIS DOWN!
```

Comment:
Unsolved TODO, not sure if it's for this PR or to be solved some time in the future.

Reply:
Good call out! I'll create a backlog item for it.